ext/src/Cassandra/Types/Float.c

```diff
@@ -0,0 +1,446 @@
+
+    result->value = self->value * flt->value;
+  } else {
+    INVALID_ARGUMENT(num, "a Cassandra\\Types\\Float");
+  }
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::div() */
+PHP_METHOD(Float, div)
+{
+  zval* num;
+
+  if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "z", &num) == FAILURE) {
+    return;
+  }
+
+  if (Z_TYPE_P(num) == IS_OBJECT &&
+      instanceof_function(Z_OBJCE_P(num), cassandra_float_ce TSRMLS_CC)) {
+    cassandra_float* self =
+        (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+    cassandra_float* flt =
+        (cassandra_float*) zend_object_store_get_object(num TSRMLS_CC);
+
+    object_init_ex(return_value, cassandra_float_ce);
+    cassandra_float* result =
+        (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+
+    if (flt->value == 0) {
+      zend_throw_exception_ex(cassandra_divide_by_zero_exception_ce, 0 TSRMLS_CC, "Divide by zero");
+      return;
+    }
+
+    result->value = self->value / flt->value;
+  } else {
+    INVALID_ARGUMENT(num, "a Cassandra\\Types\\Float");
+  }
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::abs() */
+PHP_METHOD(Float, abs)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = fabsf(self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::neg() */
+PHP_METHOD(Float, neg)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = -self->value;
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::sqrt() */
+PHP_METHOD(Float, sqrt)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  if (self->value < 0) {
+    zend_throw_exception_ex(cassandra_range_exception_ce, 0 TSRMLS_CC,
+                            "The operation would result in a complex number");
+  }
+
+  object_init_ex(return_value, cassandra_float_ce);
+  cassandra_float* result =
+      (cassandra_float*) zend_object_store_get_object(return_value TSRMLS_CC);
+  result->value = sqrtf(self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::toInt() */
+PHP_METHOD(Float, toInt)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  RETURN_LONG((long) self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::toDouble() */
+PHP_METHOD(Float, toDouble)
+{
+  cassandra_float* self =
+      (cassandra_float*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  RETURN_DOUBLE((double) self->value);
+}
+/* }}} */
+
+/* {{{ Cassandra\Types\Float::minValue() */
+PHP_METHOD(Float, minValue)
```

Comment:
I like these methods, I've seed similar ones in other classes, what do you think about making the static? E.g.:

```php
<?php
$maxFloat = Cassandra\Float::max();
```